### PR TITLE
Massive overhaul of python-soap

### PR DESCRIPTION
**Pull request by bdmartin**
## Description
Massive overhaul of python-soap to drop py2 support and add support for py38, py39, py310, py311.

See also this PR, where the changes are consumed: https://github.com/Panopto/panopto-nginx-rtmp/pull/102

## Testing
`tox run-parallel`

### CODE REVIEW

**Review by danshank — COMMENTED** [review]
All in all, looks good! Would just like some clarification before explicit approval, thank you!

File: requirements.py310.txt [hunk 1]
```diff
@@ -0,0 +1,117 @@
+#
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
+#
+#    pip-compile --no-emit-index-url --no-emit-trusted-host --output-file=requirements.py310.txt requirements-dev.txt
+#
+-e file:///C:/GitHub/python-soap
```
[the remark this answers is not in the document]
Comment (bdmartin) [reply requirements.py310.txt:7]: oops, that was supposed to be removed.

done now.

File: setup.py [hunk 2]
```diff
@@ -14,14 +12,15 @@ def readme():
...
     author='Mark Brewster',
     author_email='mbrewster@panopto.com',
-    description=('Panopto API client that wraps the zeep library for the heavy lifting'),
+    description='Panopto API client that wraps the zeep library for the heavy lifting',
     long_description=readme(),
     keywords=['python', 'panopto', 'lambda', 'api', 'soap'],
     install_requires=[
-        're',
-        'urllib3',
+        # zeep -> lxml. lxml 5.0.0 is not compatible with Python 3.9 or lower.
```
[the remark this answers is not in the document]
Comment (bdmartin) [reply setup.py:19]: I honestly don't recall.  It is probably just that pre-built wheels are not available?  So then it tries to fall back to building from source, which is a pain to get set up right on Windoze.  I vaguely recall fixing it just through trial and error.

File: tox.ini [hunk 3]
```diff
@@ -1,44 +1,65 @@
...
-alwayscopy = True
 setenv =
     # Set a stable hash seed
+    COVERAGE_FILE = {env:COVERAGE_FILE:{toxworkdir}/.coverage.{envname}}
+    PIP_DISABLE_PIP_VERSION_CHECK = 1
     PYTHONHASHSEED=0
-    COVERAGE_FILE = .coverage.{envname}
 deps =
-    -rrequirements.txt
+    -rrequirements.{envname}.txt
```
[the remark this answers is not in the document]
Comment (bdmartin) [reply tox.ini:14]: Yes, this works just fine:
```
(python-soap) [BMARTIN-DESKTOP] C:\GitHub\python-soap [bmartin/m/update-and-upgrade-python-soap ≡]
> tox run-parallel
pep8: OK ✔ in 0.75 seconds
pylint: OK ✔ in 6.52 seconds
clean: OK ✔ in 33.19 seconds
⠼ [4/8] py38 | py39 | py310 | py311C:\Users\bmartin\Envs\python-soap\lib\site-packages\setuptools\_distutils\cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
py38: OK ✔ in 4.73 seconds
py311: OK ✔ in 5.11 seconds
py39: OK ✔ in 5.12 seconds
py310: OK ✔ in 5.22 seconds
  clean: OK (33.19=setup[32.89]+cmd[0.30] seconds)
  py38: OK (4.73=setup[2.52]+cmd[0.34,1.88] seconds)
  py39: OK (5.12=setup[2.80]+cmd[0.41,1.92] seconds)
  py310: OK (5.22=setup[2.92]+cmd[0.45,1.84] seconds)
  py311: OK (5.11=setup[2.89]+cmd[0.39,1.83] seconds)
  report: OK (9.39=setup[0.00]+cmd[0.33,2.31,3.59,3.16] seconds)
  pep8: OK (0.75=setup[0.14]+cmd[0.61] seconds)
  pylint: OK (6.52=setup[3.67]+cmd[2.84] seconds)
  congratulations :) (48.12 seconds)
  ```
  But I can make the change all the same.

**Review by danshank — APPROVED** [review]

**Pull request merged**